contracts/p1/RToken.sol

```diff
@@ -274,6 +274,7 @@ contract RTokenP1 is ComponentP1, ERC20PermitUpgradeable, IRToken {
     //   totalSupply' = totalSupply - amtRToken
     function melt(uint256 amtRToken) external notPausedOrFrozen {
         _burn(_msgSender(), amtRToken);
+        require(totalSupply() >= FIX_ONE, "rToken supply too low to melt");
```

Comment:
I was imagining the require being before the burn rather than after. Do we know a reason for it to be one way vs another?